app_dart/lib/src/request_handlers/scheduler/scheduler_request_subscription.dart

```diff
@@ -59,38 +61,90 @@ class SchedulerRequestSubscription extends SubscriptionHandler {
           // we make changes for all instances.
-          final List<bbv2.BatchRequest_Request> requestListCopy = [];
-          requestListCopy.addAll(requestsToRetry);
+          final requestListCopy = [...requestsToRetry.retries];
           batchRequest.requests.clear();
           batchRequest.requests.addAll(requestListCopy);
-
-          unscheduledBuilds = requestsToRetry.map((e) => e.scheduleBuild.builder).toString();
-          if (requestsToRetry.isNotEmpty) {
-            throw InternalServerError('Failed to schedule builds: $unscheduledBuilds.');
+          errors
+            ..clear()
+            ..addAll(requestsToRetry.errors);
+          if (requestListCopy.isNotEmpty) {
+            unscheduledWarning = '${requestsToRetry.retries.map((e) => e.scheduleBuild.builder)}';
+            throw InternalServerError('Failed to schedule builds: $unscheduledWarning.');
           }
         },
         retryIf: (Exception e) => e is InternalServerError,
       );
     } catch (e) {
-      log.warning('Failed to schedule builds on exception: $unscheduledBuilds.');
-      return Body.forString('Failed to schedule builds: $unscheduledBuilds.');
+      log.warning('Failed to schedule builds on exception: $unscheduledWarning.');
+
+      await _failUnscheduledCheckRuns(batchRequest, errors);
+
+      return Body.forString('Failed to schedule builds: $unscheduledWarning.');
     }
 
     return Body.empty;
   }
 
+  Future<void> _failUnscheduledCheckRuns(bbv2.BatchRequest batchRequest, List<String> errors) async {
+    for (var failed in batchRequest.requests) {
+      final url = failed.scheduleBuild.properties.fields['git_url']?.stringValue;
+      final builder = failed.scheduleBuild.builder.builder;
+      final checkRunId = failed.scheduleBuild.tags.firstWhereOrNull((t) => t.key == 'github_checkrun')?.value;
+      final pathSegments = Uri.tryParse(url ?? '')?.pathSegments ?? [];
+      if (url == null || pathSegments.length != 2 || int.tryParse(checkRunId ?? '') == null) {
+        log.warning('missing url / github_checkrun for $builder');
+        continue;
+      }
+
+      final slug = RepositorySlug(pathSegments[0], pathSegments[1]);
+      final githubService = await config.createGithubService(slug);
+      final CheckRun checkRun = CheckRun.fromJson({
+        'id': int.parse(checkRunId!),
+        'status': CheckRunStatus.completed,
+        'check_suite': const {'id': null},
+        'started_at': '${DateTime.now()}',
+        'conclusion': null,
+        'name': builder,
+      });
+
+      await githubService.updateCheckRun(
+        slug: slug,
+        checkRun: checkRun,
+        status: CheckRunStatus.completed,
+        conclusion: CheckRunConclusion.failure,
+        output: CheckRunOutput(
+          title: 'Failed to schedule build',
+          summary: '''
+Failed to schedule `$builder`:
+
+```
+${errors.isEmpty ? 'unknown' : errors}
```

Comment:
errors.join('\n') or something might read cleaner

Reply:
It already outputs with a newline (see tests) - but having section headers might be nice.  Will do on Monday.